Lib/multiprocessing/popen_spawn_win32.py

```diff
@@ -87,12 +89,21 @@ def __init__(self, process_obj):
                                            (self.sentinel, int(rhandle)))
 
             # send information to child
-            set_spawning_popen(self)
-            try:
-                reduction.dump(prep_data, to_child)
-                reduction.dump(process_obj, to_child)
-            finally:
-                set_spawning_popen(None)
+            # bpo-42178: write to BytesIO to check for pickler exceptions
+            with io.BytesIO() as f:
+                set_spawning_popen(self)
+                try:
+                    reduction.dump(prep_data, f)
+                    reduction.dump(process_obj, f)
+                except:
+                    _winapi.TerminateProcess(self._handle)
+                else:
+                    f.seek(0)
+                    to_child.write(f.read())
```

Comment:
The thread needs to be resumed before writing to the pipe. Writing to the pipe will block if it fills up, and by default `CreatePipe` uses a 4 KiB buffer, i.e. a write that exceeds 4096 bytes will block. That's not an argument to specify a larger buffer size when creating the pipe. We just need the child to be running, so it can steal `rhandle` and read from its end.

Also, there's no need to seek the `BytesIO` instance. Just use `f.getvalue()` instead of `f.read()`.